content/How to customize the display name of ActiveAdmin search filters- Juan Ferrari.md

```diff
@@ -0,0 +1,79 @@
+
+### Before start
+
+We have to include in our `Gemfile` the `activeadmin` gem and `activeadmin-addons` gem.
+
+In this example we are going to use a search select to filter people and search a person, our `person` model has two fields named `first_name` and `last_name`, and we want to display the full name of the person when we are searching it. The `full_name` is the concatenation of `first_name` and `last_name`.
+
+## Let's do it
+
+### Create a Search Select
+
+First of all, you should create a search select input in ActiveAdmin. This can be used in many places but in this example, we are going to use it in a filter. To do that, go to an ActiveAdmin view and put this code on it:
+
+```ruby
+      filter :person_id,
+             as: :search_select_filter,
+             fields: %i[first_name last_name],
+             display_name: :full_name
+```
+
+(If you are going to use this on a form you have to change `search_select_filter` to `search_select`).
+
+The fields array indicates the fields that are used to filter, our input will be matched with the `first_name` of a person or their `last_name`.
+
+The `display_name` is the field that our filter is showing to us meanwhile we search.
+
+### Setting up the returned JSON
+
+The search select input is requesting a JSON meanwhile the user is searching for a person, for example in this case the filter will be doing requests to this endpoint `/admin/people` and retrieving a JSON showing all model attributes from a person. This is auto-generated by ActiveAdmin, but we can overwrite the response.
+
+In this application we are using JBuilder to render JSON response so
+we have to create a file in:
+
+ `views/admin/(model_name_pluralized)/index.json.jbuilder`
+
+In our example, the view created is in 
+
+`views/admin/people/index.json.jbuilder `
+
+In this file we can write the custom JSON that we want to retrieve for example:
+
+```json
+{
+  "id": person.id,
+  "full_name": person.full_name
+}
+```
+
+Attention: `full_name` is a method on `person` model, we have to write that method in the model and not in a decorator because if we are using this `search_select` in edit/new view an error will raise when validation fails, ActiveAdmin will call a method from `person` model named `full_name` and not from the decorator.
+
+### Writing the full_name method
+
+Just write the method that we want in `app/models/person.rb`:
+
+```ruby
+def full_name
+    [first_name, last_name].compact.join(' ')
+end
+```
+And now our search select input is ready to display the person full name!
+
+## Summary
+
+In this article, we learned how to display a custom field in our search select inputs. This will help us to give better search feedback to the user and it can be used in many different scenarios. It also helps us to take the control on the JSON returned by ActiveAdmin to the input (by default, all attributes are returned and maybe that's not good for us).
```

Comment:
```suggestion
In this article, we learned how to display a custom field in our search select inputs. This will help us to give better search feedback to the user and it can be used in many different scenarios. It also helps us take control of the JSON returned by ActiveAdmin to the input (by default, all attributes are returned and maybe that's not good for us).
```